### ISSUE TITLE
[feature] Display last X builds

### ISSUE THREAD

**Issue by barnabyaaron** [jan-molak/jenkins-build-monitor-plugin#252]
Can we have a settings / changes so that we can see the last X amount of builds.
I love this plugin but the problem is when we have loads of jobs to keep track of it almost becomes impossible to display.

What would be great is within the selected jobs it only displays the last x amount of builds with the status of the builds and when something new is built it's added to the start of the list and the last item is removed.

Thanks

**Comment by jan-molak** [comment]
Hey @barnabyaaron and thanks for your feedback, I'm glad that you're finding Build Monitor useful :-)

Could you please elaborate a bit more on what you have in mind? Would this be like a "filter"? So that instead of say 50 jobs you'd normally display it would only display the 10 most recent ones? Do I understand that correctly?

Best,
Jan

**Comment by barnabyaaron** [comment]
Yes so currently it is displaying all the jobs that I have filtered to display  (even if NO builds are even in this job).

What would be great is if there was a view where it would show the last x amount of builds from the selected jobs.

This means if I'm viewing 50 jobs I will only see the 10 latest builds for example (which means I can easily see this on 1 monitor)

**Comment by jan-molak** [comment]
Right, so something like "Show only 10 recently built jobs"? 
What if there are problems with any of the remaining 40 (assuming they have some builds) ?  Would that information not be important?

**Comment by barnabyaaron** [comment]
The idea is we have a monitor that we display in the office.
We have currently around 30 jobs but this will grow to over 100+ jobs over time (as we start adding more applications).
However there are normally builds only happening every say 1 hour or so (because of our process of git > review > Jenkins)

So if there can be an option so I can display the last x amount of recently built jobs this means I can easily see from the monitor if something that recently built has failed.

as a extra maybe there could be a toolbar at the bottom of this view that shows total number of jobs (that has been selected from the current filter) and how many currently have passed there latest build or failed the latest build.

**Comment by barnabyaaron** [comment]
Just a small update to this.

I have recently been using the option to 'Order By - Last build time'
which defiantly does help because it means the top left build is the latest everytime.


So technically all I need it a way to limit the build monitor window to only display the top X results because that way it will only show the last x latest builds because of the ordering of the filter.

**Comment by daantimmer** [comment]
Any progress/more thoughts on this? I too was looking for a option like this. In our case we have one generic job setup to build pull-requests and push-notifications from a TFS-Git instance. However, currently the build monitor plugin will only show the last build, while I would like to show the last-x builds.

**Comment by sflothow-vwd** [comment]
+1
I've got a large number of multi-branch pipelines, and since the Build Monitor Plugins crams all jobs into the viewport, the individual jobs become way too small. So a config option like "Show _N_ most-recently built jobs only" would be great.

**Comment by vivekquit** [comment]
Please add this feature. I am waiting to show the recent x builds and its status in the same screen

**Comment by Kregap** [comment]
+1
I have exactly the same problem as one mentioned by @sflothow-vwd where I want to list only recent jobs in multibranch pipeline configuration as now cards are getting smaller and smaller.

**Comment by nairouz12** [comment]
Hi all,
I'm facing the same issue as well.
Actually, I have a certain number of jobs (job1, job2, ..) that got executed many times (#build1, #build2,...)
I want to show the status of x number of builds per job. but the build monitor plugin shows only the latest build from each job.
this is what i would like to have:
           build#1   build#2   build#3  ...
job1     status    status      status
job2    status    status      status
job3    status    status      status
job4    status    status      status

*status, it will be showing a red or green box to indicate if the job failed or passed
Anyone who can help please?

**Comment by dcendents** [comment]
It looks like this issue is being used to ask for 2 different features:

1. Only show the latest N jobs that have built
    - To limit the number of jobs on the screen
2. Show N builds per job
    - To be able to see the status of previous builds of a job

For Feature 1, you could use the View jobs filter plugin (https://wiki.jenkins.io/display/JENKINS/View+Job+Filters) to limit the number of jobs on the page. Specifically this filter seems to do just that: https://wiki.jenkins.io/display/JENKINS/View+Job+Filters#ViewJobFilters-FiltertoMostRecentJobs

For Feature 2, I can't think of a way to do that. Myself, I normally only care for the latest build, anything before that is just volatile data of previous commits. Since I can't understand why you want that feature, it's hard for me to think of an alternate solution that would help you.
